### ISSUE TITLE
ModuleNotFoundError: No module named 'setuptools_scm'

### ISSUE THREAD

**Issue by awaemmanuel** [pypa/setuptools-scm#359]
`pip install -e .` fails with ModuleNotFoundError: No module named 'setuptools_scm'.

```
   File "/home/useradmin/sources/nlp/utils/__init__.py", line 4, in <module>
      from setuptools_scm import get_version
  ModuleNotFoundError: No module named 'setuptools_scm'
  ----------------------------------------
ERROR: Command errored out with exit status 1: /home/useradmin/miniconda3/envs/nlp_gpu/bin/python /home/useradmin/miniconda3/envs/nlp_gpu/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmplmyt1o_f Check the logs for full command output.
```
 I have setuptools_scm installed in my virtual environment. If I start a new python session from anywhere in the environment and import setuptools_scm it works fine, but pip is failing to install this package. 

**setup.py**
```
setup(
    name="utils",
    description="Utils package",
    long_description="%s\n%s"
    % (
        re.compile("^.. start-badges.*^.. end-badges", re.M | re.S).sub("", read("README.md")),
        re.sub(":[a-z]+:`~?(.*?)`", r"``\1``", read("CONTRIBUTING.md")),
    ),
    packages=["utils"],
    include_package_data=True,
    zip_safe=True,
    classifiers=[
        # complete classifier list: http://pypi.python.org/pypi?%3Aaction=list_classifiers
        "Development Status :: 5 - Production/Stable",
        "Intended Audience :: Developers",
        "License :: OSI Approved :: MIT License",
        "Operating System :: Unix",
        "Operating System :: POSIX",
        "Operating System :: Microsoft :: Windows",
        "Programming Language :: Python :: 3.6",
        "Programming Language :: Python :: 3.7",
        "Programming Language :: Python :: Implementation :: CPython",
        "Programming Language :: Python :: Implementation :: PyPy",
        "Topic :: Text Processing :: Linguistic",
        "Topic :: Utilities",
        "Intended Audience :: Science/Research",
        "Intended Audience :: Developers",
        "Intended Audience :: Education",
        "Intended Audience :: Financial and Insurance Industry",
        "Intended Audience :: Healthcare Industry",
        "Intended Audience :: Information Technology",
        "Intended Audience :: Telecommunications Industry",
    ],
    project_urls={
        "Documentation": "https://github.com/microsoft/nlp/",
        "Issue Tracker": "https://github.com/microsoft/nlp/issues",
    },
    python_requires=">=3.6",
    install_requires=["setuptools_scm>=3.2.0"],
    dependency_links=[],
    extras_require={},
    use_scm_version={"root": ".", "relative_to": __file__},
    setup_requires=["setuptools_scm"],
)
```

**Comment by RonnyPfannschmidt** [comment]
a few  things are missing here 

a) its not clear/demonstrated where the actual import to the package is happening(and `setup.py` was not pasted complete)
b) one should never use setuptools_scm inside a package (as a installed package wont have the metadata availiable)

**Comment by RonnyPfannschmidt** [comment]
based on what was pasted, i believe you have a incomplete `pyproject.toml` and import details that fetch setuptools_scm before processing setup_requires

that combined with build isolation results in setup.py running in a virtualenv that does not have setuptools_scm available

**Comment by awaemmanuel** [comment]
@RonnyPfannschmidt Thanks for your response. I'll look into your `pyproject.toml` suggestion. I currently don't have one and that may be the issue. Are there any sample `toml` files or examples to follow? I followed the instructions to setup `setuptools_scm` and I found the documentation a bit hard to follow. Can you please advise?

**Comment by RonnyPfannschmidt** [comment]
I'm currently traveling to a family vacation, so it will be a while before I can return to this

**Comment by RonnyPfannschmidt** [comment]
https://github.com/pypa/setuptools_scm#setupcfg has a minimal example of pyproject.toml at the end of the section

**Comment by RonnyPfannschmidt** [comment]
@awaemmanuel for now i presume this one is resolved, if that's in error please tell me

**Comment by jamesbraza** [comment]
For what it's worth, I ran into this today.  I also previously had `setuptools_scm[toml]` installed into my virtual environment and was confused at the error message.  The comment above put me on the scent:

> i believe you have a incomplete `pyproject.toml` and import details that fetch setuptools_scm before processing setup_requires

I realized I had forgotten to add the [suggested `pyproject.toml` section](https://github.com/pypa/setuptools_scm#pyprojecttoml-usage).

```toml
[build-system]

# SEE: https://github.com/pypa/setuptools_scm/#pyprojecttoml-usage
requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4"]
```

Hope this helps someone else out there.
